### DISCUSSION TITLE
请问怎么调整攻读学位期间发表论文与研究成果清单的顺序

### DISCUSSION THREAD

**Discussion by richie-live** [BITNP/BIThesis#407] [Q&A] [answered]
如题

**Comment by YDX-2147483647** [comment]
~~请问是要调整这两页的顺序，还是要调整每个列表中各项的顺序？~~

~~若是前者，在`main.tex`中换几个`\input{…}`中的顺序即可。若是后者，[国标](https://lib.tsinghua.edu.cn/wj/GBT7714-2015.pdf)10.1.1条规定“按正文中引用的文献出现的先后顺序连续编码”，真要改的话我可以查查。~~

（我一开始看错了）

**Reply by richie-live** [reply]
您好，我想要调整每个列表中各项的顺序，比如把比较重要的成果放前面

**Reply by YDX-2147483647** [reply, marked as answer]
> [!CAUTION]
>
> > （十二）攻读学位期间发表的论文与研究成果清单  
> >
> > 应列出攻读学位期间发表的与学位论文内容相关的学术论文和研究成果，**按发表的时间顺序列出**，研究成果可以是在学期间参加的研究项目、获得专利、获奖情况等。
>
> 如果确定要改，请往下看。

关于“攻读……清单”，我看了一下 biblatex 手册，编辑`graduate-thesis/reference/pub.bib`，加上字段`sortkey`，可以控制顺序。

```diff
@article{myCiteKey2,
  title = {交联型与线形水性聚氨酯的形状记忆性能比较},
  author = {李杰 and 张三 and 罗运军},
  author+an = {2:myself="\Author[][][共同二作]"},
  journal = {化工进展},
  year = {2007},
  number = {01},
  pages = {78-81},
  keywords = {交联;线形;水性聚氨酯;形状记忆性能},
  abstract = {由聚酯多元醇、2,4–甲苯二异氰酸酯、二羟甲基丙酸等合成了水性聚氨酯乳液,
              对乳液性能进行了研究;对制得的聚氨酯胶膜进行了红外分析、dsc测试和形状记忆性能测试;用丙三醇和三羟甲基丙烷作为扩链制得的交联水性聚氨酯与用1
              ,4–丁二醇制得的线形水性聚氨酯进行了性能比较,结果表明},
  issn = {1000-6613},
  annotation = {EI 收录, 检索号:06129773469. 已刊出.},
+ sortkey={002},
}
```

例如，上面这样可保证它在`sortkey = {001}`那项后面，且在没标`sortkey`的其它项前面。


## 更全面的例子

如果在`reference/pub.bib`中定义以下三项并引用（其它项不指定`sortkey`），那么可以保证`wandering-earth`、`wandering-earth-2`、`three-body`依次出现在清单开头，且其它项在后面按发表时间排序。

```bibtex
@misc{wandering-earth, …, sortkey = {1}, …}
@misc{wandering-earth-2, …, sortkey = {2}, …}
@misc{three-body, …, sortkey = {13}, …}
```

**Reply by richie-live** [reply]
可以啦，感谢大佬！

**Comment by qiu-wanyong** [comment]
如果想要按照引用顺序排序，可以开启 `publications/sorting` 选项。这个在哪设置？

**Reply by YDX-2147483647** [reply]
你好！在`main.tex`里的`\BITSetup{…}`中设置，大概下面这样。

```LaTeX
\BITSetup{
  …,
  publications/sorting  = {false},
}
```

**Comment by YDX-2147483647** [comment]
为方便大家查看，这个问题归档到了 [🥑 疑难杂症 (bithesis.bitnp.net/faq)](https://bithesis.bitnp.net/faq.html)，后续将在那边更新。如果遇到新问题，欢迎继续在 GitHub Discussions 讨论。

https://bithesis.bitnp.net/faq/bib-sort.html
https://github.com/BITNP/BIThesis-wiki/blob/96df7d463a26ed01a2000ee13e2a3b6c4f5c2452/wiki/faq/bib-sort.md
